src/Notes/TokenExchangeFailure.php

```diff
@@ -74,4 +75,21 @@ public static function possibly_add_note() {
 		$note->save();
 		return true;
 	}
+
+	/**
+	 * Set note to actioned.
+	 *
+	 * @since x.x.x
+	 *
+	 * @return void
+	 */
+	public static function possibly_action_note() {
+		if ( ! self::note_exists() ) {
```

Comment:
I had to do few tests: success case when sooner or later retry will work out, the case when I click on a banner's action button and the case if all retry actions fail. 

I could not see the note after the first test case because note is in the database but actioned. What if we change note check to check for a note and a status instead. This way if a user will install and rollback multiple times the banner will show again.

```suggestion
		if ( ! self::note_exists() && ! self::has_note_been_actioned() ) {
```

Reply:
I really don't want to support rolling back multiple times. You never know what can happen.

In `possibly_action_note` we want to hide note in case the procedure was successful. With this change the note will never be hidden unless somebody will click the button on the notice.

Reply:
Successful scheduled action execution will also hide it. But ok, maybe my case is synthetic, and we do not need this.